src/application/application-scopes/application-scopes.service.ts

```diff
@@ -0,0 +1,141 @@
+import {
+  BadRequestException,
+  HttpException,
+  HttpStatus,
+  Injectable,
+  InternalServerErrorException,
+  Logger,
+} from '@nestjs/common';
+import { randomUUID } from 'crypto';
+import { ScopeDto, UpdateScopeDto } from 'src/dto/application.dto';
+import { PrismaService } from 'src/prisma/prisma.service';
+
+@Injectable()
+export class ApplicationScopesService {
+  private readonly logger: Logger;
+  constructor(private readonly prismaService: PrismaService) {
+    this.logger = new Logger();
+  }
+
+  async createScope(data: ScopeDto, applicationsId: string, scopeId?: string) {
+    const application = await this.prismaService.application.findUnique({
+      where: { id: applicationsId },
+    });
+    if (!application) {
+      throw new HttpException(
+        'Application with provided id dont exist',
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+
+    const id = data.id ? data.id : scopeId ? scopeId : randomUUID();
```

Comment:
Please only use **?** Operators in single condition checks. Separate into if else blocks to keep it simpler